File: DataOP/ConnectToCRM.cs
using System;
using System.Net.Http;

namespace zoiper_sdk.DataOP
{
    class ConnectToCRM
    {
        public static HttpClient TheHttpClient(string apiVersion)
        {
            Configuration config = new Configuration("CrmOnline");
            Authentication auth = new Authentication(config);

            string url = config.ServiceUrl;

            //string url = new Configuration("CrmOnline").ServiceUrl.ToString();

            string accessToken = Authentication.AcquireToken().AccessToken;

            HttpClient httpClient = new HttpClient();
            httpClient.BaseAddress = new Uri(url + apiVersion);
            httpClient.Timeout = new TimeSpan(0, 2, 0);
            httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", accessToken);
            httpClient.DefaultRequestHeaders.Add("OData-MaxVersion", "4.0");
            httpClient.DefaultRequestHeaders.Add("OData-Version", "4.0");
            httpClient.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
            return httpClient;
        }
    }
}

File: OzekiDemo/Service/DynamicDataService/ConnectToCRM.vb
Imports System.Net.Http

Public Class ConnectToCRM
    Sub New()

    End Sub
    Function TheHttpClient(ByVal apiVersion As String) As HttpClient
        Dim config As Configuration = New Configuration("CrmOnline")
        Dim auth As Authentication = New Authentication(config)
        Dim url As String = config.ServiceUrl
        Dim accessToken As String = Authentication.AcquireToken().AccessToken
        Dim httpClient As HttpClient = New HttpClient()
        httpClient.BaseAddress = New Uri(url & apiVersion)
        httpClient.Timeout = New TimeSpan(0, 2, 0)
        httpClient.DefaultRequestHeaders.Authorization = New System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", accessToken)
        httpClient.DefaultRequestHeaders.Add("OData-MaxVersion", "4.0")
        httpClient.DefaultRequestHeaders.Add("OData-Version", "4.0")
        httpClient.DefaultRequestHeaders.Accept.Add(New System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"))
        Return httpClient
    End Function
End Class
